region/maridia/inner-green/Oasis.json

```diff
@@ -203,6 +173,88 @@
         }
       }
     },
+    {
+      "link": [1, 1],
+      "name": "Suitless Water Shinecharge, Shinespark Return",
+      "notable": false,
+      "entranceCondition": {
+        "comeInRunning": {
+          "speedBooster": true,
+          "minTiles": 2.4375
+        }
+      },
+      "requires": [
+        "canStutterWaterShineCharge",
+        "canShinechargeMovementComplex",
+        {"canShineCharge": {
+          "usedTiles": 33,
+          "openEnd": 2
+        }},
+        {"shinespark": {"frames": 7}}
+      ],
+      "exitCondition": {
+        "leaveWithSpark": {}
+      },
+      "note": "Get the shinecharge then move towards and shinespark out the left door."
+    },
+    {
+      "link": [1, 2],
+      "name": "Leave Shinecharged",
+      "notable": false,
+      "entranceCondition": {
+        "comeInShinecharging": {
+          "length": 13,
+          "openEnd": 0
+        }
+      },
+      "requires": [
+        "Gravity"
+      ],
+      "exitCondition": {
+        "leaveShinecharged": {
+          "framesRemaining": 179
+        }
+      }
+    },
+    {
+      "link": [1, 2],
+      "name": "Suitless Water Shinecharge",
+      "notable": false,
+      "entranceCondition": {
+        "comeInRunning": {
+          "speedBooster": true,
+          "minTiles": 2
+        }
+      },
+      "requires": [
+        "canStutterWaterShineCharge",
+        {"canShineCharge": {
+          "usedTiles": 33,
+          "openEnd": 2
+        }}
+      ],
+      "exitCondition": {
+        "leaveShinecharged": {
+          "framesRemaining": 179
```

Comment:
Leaving with blue is trivial from here but hits some schema problems.
- Coming from a water room without gravity, with speed
- Low run speed after running long enough to gain blue

I think we mainly lose temp blue strats, and things like `canBlueSpaceJump`